[feat] Add support to wallpaper module for multiple screens

anihm136 opened the pull request:
This PR adds support to the wallpaper module for multiple screens. It introduces the following changes -
1. Calling `setup` multiple times with different `screen` arguments works correctly
2. A new `screens` argument to the setup function, which allows setup of multiple screens by passing a table of screens. Currently only supports the simple and random pickers

JavaCafe01 commented:
@anihm136 I don't think we need two parameters for screen. We can have one parameter, and check if it's an array or not, and deal with it accordingly underneath. @Nooo37 is that fine or do you have any other ideas?

Nooo37 commented:
I thought about that too. If it comes so naturally to mind, we should probably do it

anihm136 commented:
Sounds good. It should make it easier to handle as well. I was just following the awful.wallpaper API initially, but having a single parameter makes more sense. Will update

anihm136 commented:
I have updated the wallpaper module to support passing either a screen object or a table of screen objects as the `screen` parameter. Updated the docs to reflect the same. Please go through and let me know if it works as expected and if any other updates are required

Nooo37 commented on module/wallpaper.lua:
```diff
@@ -154,7 +158,15 @@ local simple_index = 0
     simple_index = (simple_index % #wallpapers) + 1
-    apply(wallpapers[simple_index], args)
+    if type(args.screen) == 'table' then
+        for _,v in ipairs(args.screens) do
```
Pretty sure you have to check if args.screens is a table here? Doesn't make a lot of sense to check if args.screen is a table and then iterate over args.screens imo. You did that in other parts of the code too, or am I missing something

anihm136 replied:
Good catch. The check is supposed to be for `args.screen`, since we are eliminating `args.screens`. Missed changing in the iteration

Nooo37 commented on module/wallpaper.lua:
```diff
@@ -310,7 +330,11 @@ function setup(args)
     local function set_wallpaper(s)
-        config.screen = s or config.screen
+        if type(config.screen) == 'table' then
+        else
```
Just do `~=` here instead of an empty `if` block

anihm136 replied:
Done, remnant of the old code

Nooo37 reviewed (changes requested):
Will have to take another look as I haven't really tried it yet sorry

Nooo37 commented:
Sorry for taking so long to come back to here. Seems to be fine now, thanks for your PR :rocket:
